src/main/org/pantsbuild/jarjar/MainProcessor.java

```diff
@@ -52,7 +54,7 @@ public MainProcessor(List<PatternElement> patterns, boolean verbose, boolean ski
         if (kp != null)
             processors.add(kp);
         processors.add(new ZapProcessor(zapList));
-        processors.add(new JarTransformerChain(new RemappingClassTransformer[]{ new RemappingClassTransformer(pr) }));
```

Comment:
What do you think about adding a new processor instead of overriding JarTransformer that runs before JarTransformer. It could identify misplace classes and decide what to do as a separate processing step.  Then you don't have to modify the JarTransformerChain API.